splash/tests/test_execute.py

```diff
@@ -2017,6 +2017,47 @@ def test_related_resources(self):
         img = Image.open(StringIO(resp.content))
         self.assertNotEqual((0,0,0,255), img.getpixel((10, 10)))
 
+    def test_disable_js(self):
+        script = """
+        function main(splash)
+            splash:disable_js()
+            splash:set_content{
+                data = [[
+                    <html><body bgcolor="#FFFFFF">
+                        <script> document.body.style.backgroundColor = "#000000"; </script>
+                    </body></html>
+                ]],
+                baseurl = splash.args.base,
+            }
+            return splash:png()
+        end
+        """
+        resp = self.request_lua(script, {"base": self.mockurl("")})
+        self.assertStatusCode(resp, 200)
+        img = Image.open(StringIO(resp.content))
```

Comment:
I think this way to check if JS is enabled is too indirect; it is better to check just the resulting HTML than to render an image and encode the information in pixel colors.